enterprise/fulltext-addon/src/test/java/org/neo4j/kernel/api/impl/fulltext/integrations/bloom/BloomIT.java

```diff
@@ -341,6 +343,41 @@ public void shouldBeAbleToRunConsistencyCheck() throws Exception
         assertTrue( result.isSuccessful() );
     }
 
+    @Test
+    public void failureToStartUpMustNotPreventShutDown() throws Exception
+    {
+        builder.setConfig( LoadableBloomFulltextConfig.bloom_indexed_properties, "prop" );
+
+        // Create the store directory and all its files, and add a bit of data to it
+        GraphDatabaseService db = builder.newGraphDatabase();
+        try ( Transaction tx = db.beginTx() )
+        {
+            db.createNode().setProperty( "prop", "bla bla bla" );
+            tx.success();
+        }
+        db.shutdown();
+
+        File dir = testDirectory.graphDbDir();
+        assertTrue( dir.setReadable( false ) );
+        try
+        {
+            // Making the directory not readable ought to cause problems for the database as it tries to start up
+            builder.newGraphDatabase().shutdown();
+            fail( "Should not have started up and shut down cleanly on an unreadable store directory" );
+        }
+        catch ( Exception e )
+        {
+            // Good
+        }
+        finally
+        {
+            if ( !dir.setReadable( true ) )
+            {
+                System.err.println( "!!! Failed to make " + dir + " writable again !!!" );
```

Comment:
Oo

Reply:
I didn't make it an assertion in case it would shadow another throwable from the try-clause.